src/idl_gen_cpp.cpp

```diff
@@ -577,12 +578,17 @@ class CppGenerator : public BaseGenerator {
     return ret;
   }
 
+  bool FlexibleStringConstructor(const FieldDef *field) {
+    auto attr = field ? (field->attributes.Lookup("cpp_str_flex_ctor") != nullptr) : false;
+    auto ret = attr ? attr : parser_.opts.cpp_object_api_string_flexible_constructor;
+    return (ret && NativeString(field) != "std::string"); // Only for custom string types.
+  }
+
   std::string GenTypeNativePtr(const std::string &type, const FieldDef *field,
                                bool is_constructor) {
     auto &ptr_type = PtrType(field);
     if (ptr_type != "naked") {
-      return (ptr_type != "default_ptr_type" ? ptr_type :
-              parser_.opts.cpp_object_api_pointer_type) + "<" + type + ">";
+      return (ptr_type != "default_ptr_type" ? ptr_type : parser_.opts.cpp_object_api_pointer_type) + "<" + type + ">";
```

Comment:
make sire all lines are <80 columns

Reply:
In the first clang-format run I did, I disabled the enforcement of 80 chars lines, mostly because as you can see from the new commit 4791923806965637d5b13f7003329bfbb2fdf18b it formats a lot of code. But now it's fully compliant with the provided clang-format file, which it wasn't before. It's a good thing, I just wasn't sure if you'd accept this kind of changes in a PR.

Reply:
Typically prefer `git clang-format` to limit changes to the new code, yes.